src/features/duckplayer/components/ddg-video-overlay.js

```diff
@@ -0,0 +1,132 @@
+import css from '../assets/video-overlay.css'
+import dax from '../assets/dax.svg'
+import { i18n } from '../text.js'
+import { appendImageAsBackground } from '../util.js'
+import { VideoOverlayManager } from '../video-overlay-manager.js'
+
+/**
+ * The custom element that we use to present our UI elements
+ * over the YouTube player
+ */
+export class DDGVideoOverlay extends HTMLElement {
+    static CUSTOM_TAG_NAME = 'ddg-video-overlay'
+    /**
+     * @param {import("../overlays.js").Environment} environment
+     * @param {import("../util").VideoParams} params
+     * @param {VideoOverlayManager} manager
+     */
+    constructor (environment, params, manager) {
+        super()
+        if (!(manager instanceof VideoOverlayManager)) throw new Error('invalid arguments')
+        this.environment = environment
+        this.params = params
+        this.manager = manager
+
+        /**
+         * Create the shadow root, closed to prevent any outside observers
+         * @type {ShadowRoot}
+         */
+        const shadow = this.attachShadow({ mode: this.environment.isTestMode() ? 'open' : 'closed' })
+
+        /**
+         * Add our styles
+         * @type {HTMLStyleElement}
+         */
+        const style = document.createElement('style')
+        style.innerText = css
+
+        /**
+         * Create the overlay
+         * @type {HTMLDivElement}
+         */
+        const overlay = this.createOverlay()
+
+        /**
+         * Append both to the shadow root
+         */
+        shadow.appendChild(overlay)
+        shadow.appendChild(style)
+    }
+
+    /**
+     * @returns {HTMLDivElement}
+     */
+    createOverlay () {
+        const overlayElement = document.createElement('div')
+        overlayElement.classList.add('ddg-video-player-overlay')
+        overlayElement.innerHTML = `
```

Comment:
Let's fast follow with that super lightweight HTML escaped lib that we talked about... please ping me to find it once ready.

Reply:
https://github.com/mozilla-extensions/secure-proxy/blob/db4d1b0e2bfe0abae416bf04241916f9e4768fd2/src/commons/template.js#L62 << this code